tests/Doctrine/Tests/ORM/Functional/Ticket/GH11037/GH11037Test.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket\GH11037;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Tools\SchemaValidator;
+use Doctrine\Tests\OrmTestCase;
+
+/**
+ * @requires PHP >= 8.1
+ */
+final class GH11037Test extends OrmTestCase
+{
+    /** @var EntityManagerInterface */
+    private $em;
+
+    /** @var SchemaValidator */
+    private $validator;
+
+    protected function setUp(): void
+    {
+        $this->em        = $this->getTestEntityManager();
+        $this->validator = new SchemaValidator($this->em);
+    }
+
+    public function testMetadataFieldTypeCoherentWithEntityPropertyType(): void
+    {
+        $class = $this->em->getClassMetadata(ValidEntityWithTypedEnum::class);
+        $ce    = $this->validator->validateClass($class);
+
+        self::assertEquals([], $ce);
+    }
+
+    public function testMetadataFieldTypeNotCoherentWithEntityPropertyType(): void
+    {
+        $class = $this->em->getClassMetadata(InvalidEntityWithTypedEnum::class);
+        $ce    = $this->validator->validateClass($class);
+
+        self::assertEquals(
+            [
+                "The field 'Doctrine\Tests\ORM\Functional\Ticket\GH11037\InvalidEntityWithTypedEnum#status1' has the property type 'Doctrine\Tests\ORM\Functional\Ticket\GH11037\StringEntityStatus' that differs from the metadata field type 'int' returned by the 'integer' DBAL type.",
+                "The field 'Doctrine\Tests\ORM\Functional\Ticket\GH11037\InvalidEntityWithTypedEnum#status2' has the property type 'Doctrine\Tests\ORM\Functional\Ticket\GH11037\IntEntityStatus' that differs from the metadata enumType 'Doctrine\Tests\ORM\Functional\Ticket\GH11037\StringEntityStatus' returned by the 'integer' DBAL type.",
```

Comment:
I don't think that this is accurate. The DBAL has no knowledge of enums. The transformation from integer to enums does not happen in the DBAL type, but here: https://github.com/doctrine/orm/blob/609647a51a1b28a674de270204a5cea3af57d554/lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php#L701-L710

Anyway, I think the message should not mention such implementation details: the fact that the contents of `enumType` differs from the property type is enough to tell that there is something wrong.

Reply:
I wasn't sure about it, thanks! 👍